pandas/core/groupby/groupby.py

```diff
@@ -464,6 +464,17 @@ def _group_selection_context(groupby):
     groupby._reset_group_selection()
 
 
+@contextmanager
+def _observed_is_true(groupby):
```

Comment:
we should already have a temp_setattr context manager to do this (i like this name slightly better) but use the existing pattern (following PR to change the name is ok)

Reply:
Ah, cool. Switching to `com.temp_setattr`.